frontend/src/methods/machine.ts

```diff
@@ -151,8 +152,12 @@ const copyUserLabels = (src: Resource, dst: Resource) => {
 }
 
 export const updateTalosMaintenance = async (machine: string, talosVersion: string, schematic?: string) => {
+  const imageFactoryBaseURL = await getImageFactoryBaseURL();
+
+  const host = new URL(imageFactoryBaseURL).host;
+
   const image = schematic ?
-    `factory.talos.dev/installer/${schematic}:v${talosVersion}` :
+    `${host}/installer/${schematic}:v${talosVersion}` :
     `ghcr.io/siderolabs/installer:v${talosVersion}`;
```

Comment:
i guess this is also hardcoded in a way :sweat_smile:

Reply:
Yep, but probably much less of a problem than the other. Can live with it for now I think.